core/src/main/java/org/infinispan/persistence/sifs/Index.java

```diff
@@ -77,18 +82,44 @@ public Index(NonBlockingManager nonBlockingManager, FileProvider fileProvider, P
       this.maxNodeSize = maxNodeSize;
       indexDir.toFile().mkdirs();
 
+      // If there are more index files than our segments, that means we can't load them
+      boolean hasExtraIndexFile = fileExists(segments);
```

Comment:
shouldn't be `segments - 1`?
nit, invert the boolean here instead of doing every time in the loop.
```suggestion
      boolean attemptLoad = !fileExists(segments - 1);
```

Reply:
> shouldn't be `segments - 1`? nit, invert the boolean here instead of doing every time in the loop.

Nope, this is checking for an extra file which would be just segments. This covers the case that if a user lowers the number of segments that we consider the index dirty. I can try to clarify the comment.

Reply:
got it.
what would happen if the user increments the number of segments? I assume the index is still valid and the new entries would be "indexed" in the new "segments", right? 

nit, should we rename segments => buckets/shards (or something else) to avoid confusion with the cache segments?

Reply:
> I assume the index is still valid and the new entries would be "indexed" in the new "segments", right?

Nope, we check for the existence of the file and if it doesn't exist or doesn't have the GRACEFULLY shutdown bytes we assume the entire index is dirty https://github.com/infinispan/infinispan/blob/main/core/src/main/java/org/infinispan/persistence/sifs/Index.java#L263